### PR TITLE
CNS-710-providers-selfdelegation-reward

### PR DESCRIPTION
**Pull request by Yaroms**

### CODE REVIEW

File: testutil/e2e/paymentE2E.go [hunk 1]
```diff
@@ -146,7 +149,7 @@ func (lt *lavaTest) checkPayment(providers []string, startBalances []sdk.Coin) {
 	}
 
 	// get new balance and checks that at least one provider's balance was increased
-	newBalances, err := lt.getBalances(providers)
+	newBalances, err := lt.getRewards(providers)
```
Comment (oren-lava) [inline testutil/e2e/paymentE2E.go:152]: change the variable name for clarity. Those are rewards, not balances

File: x/pairing/keeper/cu_tracker_test.go [hunk 2]
```diff
@@ -249,18 +249,21 @@ func TestTrackedCuWithQos(t *testing.T) {
...
 
-			newBalance1 := ts.GetBalance(provider1Acc.Addr)
-			newBalance2 := ts.GetBalance(provider2Acc.Addr)
-
-			require.Equal(t, balance1+tt.p1ExpectedReward, newBalance1)
-			require.Equal(t, balance2+tt.p2ExpectedReward, newBalance2)
+			reward, err := ts.QueryDualstakingDelegatorRewards(provider1Acc.Addr.String(), provider1Acc.Addr.String(), ts.spec.Index)
+			require.Nil(ts.T, err)
+			require.Equal(ts.T, tt.p1ExpectedReward, reward.Rewards[0].Amount.Amount.Int64())
+			_, err = ts.TxDualstakingClaimRewards(provider1Acc.Addr.String(), provider1Acc.Addr.String())
```
Comment (oren-lava) [inline x/pairing/keeper/cu_tracker_test.go:259]: why not checking the balance after claiming? also verify that the rewards are reset after claiming?

File: x/pairing/keeper/cu_tracker_test.go [hunk 3]
```diff
@@ -249,18 +249,21 @@ func TestTrackedCuWithQos(t *testing.T) {
...
+			require.Nil(ts.T, err)
+
+			reward, err = ts.QueryDualstakingDelegatorRewards(provider2Acc.Addr.String(), provider2Acc.Addr.String(), ts.spec.Index)
+			require.Nil(ts.T, err)
+			require.Equal(ts.T, tt.p2ExpectedReward, reward.Rewards[0].Amount.Amount.Int64())
+			_, err = ts.TxDualstakingClaimRewards(provider2Acc.Addr.String(), provider2Acc.Addr.String())
```
Comment (oren-lava) [inline x/pairing/keeper/cu_tracker_test.go:265]: also here balance + reward reset checks

File: x/pairing/keeper/helpers_test.go [hunk 4]
```diff
@@ -213,9 +210,15 @@ func (ts *tester) payAndVerifyBalance(
...
-	actualReward := ts.GetBalance(providerAddr)
-	require.Equal(ts.T, expectedReward, actualReward)
+
+	reward, err := ts.QueryDualstakingDelegatorRewards(providerAddr.String(), providerAddr.String(), "")
+	require.Nil(ts.T, err)
+	for _, reward := range reward.Rewards {
+		want = want.Sub(reward.Amount.Amount)
+	}
+	require.True(ts.T, want.IsZero())
+	_, err = ts.TxDualstakingClaimRewards(providerAddr.String(), providerAddr.String())
```
Comment (oren-lava) [inline x/pairing/keeper/helpers_test.go:220]: check balance after claiming

File: x/pairing/keeper/grpc_query_effective_policy.go [hunk 5]
```diff
@@ -22,7 +22,7 @@ func (k Keeper) EffectivePolicy(goCtx context.Context, req *types.QueryEffective
 	if err != nil {
 		return nil, err
 	}
-	strictestPolicy, _, err := k.GetProjectStrictestPolicy(ctx, project, req.SpecID)
+	strictestPolicy, _, err := k.GetProjectStrictestPolicy(ctx, project, req.SpecID, uint64(ctx.BlockHeight()))
```
Comment (oren-lava) [inline x/pairing/keeper/grpc_query_effective_policy.go:25]: if you use here current height, it means that if I run the effective policy query with a different height (using the --height flag), it won't work right? (not sure, but worth a check)

File: testutil/e2e/paymentE2E.go (hunk 1 above)
Comment (Yaroms) [reply testutil/e2e/paymentE2E.go:152]: done

File: x/pairing/keeper/grpc_query_effective_policy.go (hunk 5 above)
Comment (Yaroms) [reply x/pairing/keeper/grpc_query_effective_policy.go:25]: when using --height the ctx.blockheight changes accordingly

**Review by oren-lava — APPROVED** [review]

**Pull request merged**